js/utils.js

```diff
@@ -1213,40 +1263,73 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
 
     return cachedFiles[cacheKey]
       .then(function(response) {
+        var image = _.get(data, ['record', 'data', data.field.column]);
+
         if (!data.field) {
           return data.record;
         }
 
-        var image = _.get(data, ['record', 'data', data.field.column]);
+        if (data.field.from === 'details') {
+          var imageFiles = [];
+          var images;
+          var fileExtensionRegex = /(.+?)(?:\.[^\.]*$|$)/;
+          var fileNameRegex = /[^\\\/]+$/igm;
 
-        if (_.isArray(image)) {
-          image = _.compact(image)[0];
-        }
+          if (typeof image === 'string') {
+            images = splitByCommas(image);
+          } else {
+            images = image;
+          }
+
+          _.forEach(response.files, function(file) {
+            var fileName = file.name.match(fileExtensionRegex)[1];
+
+            _.forEach(images, function(image) {
+              var imageNameFromURL = image.match(fileNameRegex);
+              var imageName = imageNameFromURL
+                ? imageNameFromURL[0]
+                : image;
+
+              if (imageName && (file.name === imageName || fileName === imageName)) {
+                imageFiles.push(file.url);
+              } else if (Static.RegExp.number.test(imageName)
+                && parseInt(imageName, 10) === file.id) {
+                imageFiles.push(file.url);
+              }
+            });
+          });
+
+          _.set(data, ['record', 'data', data.field.column], imageFiles);
+        } else {
+          if (_.isArray(image)) {
+            image = _.compact(image)[0];
+          }
 
-        if (isValidImageUrl(image)) {
+          if (isValidImageUrl(image)) {
           // Record data doesn't need updating
-          return data.record;
-        }
+            return data.record;
+          }
 
-        var urlEdited = _.some(response.files, function(file) {
+          var urlEdited = _.some(response.files, function(file) {
           // remove file extension
-          var fileName = file.name.match(/(.+?)(?:\.[^\.]*$|$)/)[1];
+            var fileName = file.name.match(fileExtensionRegex)[1];
```

Comment:
@YaroslavOvdii The `fileExtensionRegex` variable is not defined here.

Reply:
@tonytlwu It's defined at this [line](https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/562/files#diff-1e20a0ee0fa5c5689cd8541a5892e5e23248b3bf2ab918043e525c443856f18cR1275)

Reply:
@YaroslavOvdii But that's in the `if {}` clause and this is the `else {}` clause, which doesn't get access to that variable. That's why it's `undefined`.

Reply:
@tonytlwu Yes, you are right. It seems to be a problem. Because it was defined in the if clauser.